section1/reflection.md

```diff
@@ -2,18 +2,59 @@
 
 1. Regarding the blog posts in Part A, how do you feel about asking questions? Do you tend to ask them too soon, or wait too long, or somewhere in between?
 
+I feel good about asking questions. I like the advice to ask for help after 30 minutes of being stuck. Usually I
+tend to wait longer than I should to ask for help. But it seems easy and very encouraged to ask for help in the
+environment Turing creates. I realize its valuable to try to solve a problem on your own, but its wasteful to spend
+too much time working on the problem if you are stuck.
+
+
 2. What are the data types you learned about in this section? In your own words, define each.
 
+I learned about string, number, and boolean data types in this section.
+* A string is a set of letters or other characters contained within quotes. Even numbers can be in a string if they are inside the quotes.
+* A number is any positive, negative, or floating point number that defines a variable.
+* A boolean is a data type with a true or a false value.
+
+
 3. How would you log the string `"Hello World!"` to the console?
 
+```javascript
+console.log("Hello World!");
+```
+
+
 4. What is/are the character(s) you would use to indicate comments in a JavaScript file? What is the purpose of a code comment?
 
+You would use either // for mostly single line comments and /* */for multi-line comments. The purpose of code comments is to
+add information for other developers and to de-activate any code you want to keep from running.
+
+
 5. In your own words, what is a variable? How would you explain it to a 5 year old?
 
+A variable is a way to store information. It is like a box that has something in it. When you use the box the thing inside the box is accessed.
+
+
 6. Think of a site or app you use frequently. What are three variables that are probably used? Which data type would each of those variables probably hold?
 
+My email login
+
+`var username = string`
+`var logInAttempts = number`
+`var isLoggedIn = boolean`
+
+
 7. In your own words, explain what concatenation is.
 
+Concatenation is a way to join together strings and variables with different values.
+
+
 8. Think of a site or app you use frequently. Where do you think the developers used concatenation?
 
+Spotify. They used concatenation when combining information on an artist. For example it could be something like this
+
+`var paulMcartney = "${songs} + ${albums} + ${genre}";`
+
+
 9. What questions do you still have about the work we've done so far? (not having a question is not an option)
+
+What exactly is `console.log` and what is it doing when you use it? I understand the variables and how that information is integrated, but I'm unclear on exactly what putting them into `console.log` ,like we have in these exercises, does exactly. I'm sure with different examples and more practice the concept behind JS will become more clear.
```

Comment:
`console.log` is used mostly as a way to debug your code, once your code is added to a webpage instead of just a program run in your terminal, using `console.log` will be a great way to see your variables and if you have access to what you think you do on a certain page.